tests/test_type_caster_pyobject_ptr.cpp

```diff
@@ -127,4 +127,26 @@ TEST_SUBMODULE(type_caster_pyobject_ptr, m) {
         (void) py::cast(*ptr);
     }
 #endif
+
+    m.def("pyobject_ptr_from_handle_nullptr", []() {
```

Comment:
not very clear at first glance how this handle_nullptr logic is used with the rest of the change. Could you elaborate a bit more on this? Also surprising to me is that if a py::handle can already hold a nullptr, then why do we need to add another value_is_nullptr member to the arg struct, which already hold a py::handle member?

Reply:
Thanks!